templates/nft-template/move/sources/nft_launchpad.move

```diff
@@ -0,0 +1,462 @@
+        collection_owner_obj: object::Object<CollectionOwnerObjConfig>,
+    }
+
+    /// Global per contract
+    struct Registry has key {
+        collection_objects: vector<object::Object<collection::Collection>>
+    }
+
+    /// Global per contract
+    struct Config has key {
+        admin_addr: address,
+        pending_admin_addr: option::Option<address>,
+        mint_fee_collector_addr: address,
+    }
+
+    // If you deploy the module under an object, sender is the object's signer
+    // If you deploy the module under your own account, sender is your account's signer
+    fun init_module(sender: &signer) {
+        move_to(sender, Registry {
+            collection_objects: vector::empty()
+        });
+        move_to(sender, Config {
+            admin_addr: signer::address_of(sender),
+            pending_admin_addr: option::none(),
+            mint_fee_collector_addr: signer::address_of(sender),
+        });
+    }
+
+    // ================================= Entry Functions ================================= //
+
+    public entry fun set_pending_admin(sender: &signer, new_admin: address) acquires Config {
+        let sender_addr = signer::address_of(sender);
+        let config = borrow_global_mut<Config>(@launchpad_addr);
+        assert!(is_admin(config, sender_addr), ENOT_ADMIN);
+        config.pending_admin_addr = option::some(new_admin);
+    }
+
+    public entry fun accept_admin(sender: &signer) acquires Config {
+        let sender_addr = signer::address_of(sender);
+        let config = borrow_global_mut<Config>(@launchpad_addr);
+        assert!(is_pending_admin(config, sender_addr), ENOT_PENDING_ADMIN);
+        config.admin_addr = sender_addr;
+        config.pending_admin_addr = option::none();
+    }
+
+    public entry fun update_mint_fee_collector(sender: &signer, new_mint_fee_collector: address) acquires Config {
+        let sender_addr = signer::address_of(sender);
+        let config = borrow_global_mut<Config>(@launchpad_addr);
+        assert!(is_admin(config, sender_addr), EONLY_ADMIN_CAN_UPDATE_MINT_FEE_COLLECTOR);
+        config.mint_fee_collector_addr = new_mint_fee_collector;
+    }
+
+    public entry fun create_collection(
+        sender: &signer,
+        description: string::String,
+        name: string::String,
+        uri: string::String,
+        max_supply: option::Option<u64>,
+        royalty_percentage: option::Option<u64>,
+        pre_mint_amount: u64,
+        allowlist: option::Option<vector<address>>,
+        allowlist_start_time: option::Option<u64>,
+        allowlist_end_time: option::Option<u64>,
+        allowlist_mint_limit_per_addr: option::Option<u64>,
+        allowlist_mint_fee_per_nft: option::Option<u64>,
+        public_mint_start_time: option::Option<u64>,
+        public_mint_end_time: option::Option<u64>,
+        public_mint_limit_per_addr: option::Option<u64>,
+        public_mint_fee_per_nft: option::Option<u64>,
```

Comment:
Would be nice to have this in a separate function. Too many args being passed in this function I think